fishtest/utils/create_indexes.py

```diff
@@ -23,28 +23,62 @@
 pgns = db['pgns']
 
 
-indexList = { 'runs' : [ [(u'args.username', ASCENDING)],
-                         [(u'finished', ASCENDING), (u'last_updated', DESCENDING), (u'start_time', DESCENDING)] ],
-                         # [(u'finished', ASCENDING), (u'last_updated', DESCENDING)],
-                         # [(u'deleted', ASCENDING)],
-                         # [(u'deleted', ASCENDING), (u'finished', ASCENDING)],
-                         # [(u'last_updated', DESCENDING), (u'start_time', DESCENDING)],
-                         # [(u'tasks.pending', ASCENDING)],
-              'pgns' : [ [('run_id', ASCENDING)] ],
-              'users' : [ [('username', ASCENDING)] ],
-              'actions' : [ [('username', ASCENDING)] ],
-            }
-uniqueList = [ 'username' ]
+index_list = {
+  'runs' : [
+    [
+      # Index used for querying combos of username / Greens / LTC
+      ('finished', DESCENDING),
+      ('args.username', ASCENDING),
+      ('args.ltc', ASCENDING),
+      ('results_info.style', ASCENDING),
+      ('last_updated', DESCENDING),
+    ],
+    [
+      # Index used for get_unfinished_runs()
+      ('finished', ASCENDING),
+      ('last_updated', DESCENDING),
+      ('start_time', DESCENDING),
+    ],
+    # [(u'deleted', ASCENDING)],
+    # [(u'deleted', ASCENDING), (u'finished', ASCENDING)],
+    # [(u'last_updated', DESCENDING), (u'start_time', DESCENDING)],
+    # [(u'tasks.pending', ASCENDING)],
+  ],
+  'old_runs' : [
+    [
+      # Index used for querying combos of username / Greens / LTC only
+      ('finished', DESCENDING),
+      ('args.username', ASCENDING),
+      ('args.ltc', ASCENDING),
+      ('results_info.style', ASCENDING),
+      ('_id', DESCENDING),
```

Comment:
`_id` index is default

Reply:
@tomtor my bad but I don't understand the type of correction that you are suggesting:
 1. delete `('_id', DESCENDING),`
 2. write `('_id', DEFAULT)`
 3. something else

Reply:
I meant 1.

Reply:
@linrock please delete `('_id', DESCENDING),`

Reply:
there's a single-field `_id` index on collections by default yes, but it can't be used here.

`_id` needs to be on this compound index since the old_runs query filters by a few fields, then sorts by `_id`.

Try running this example code with and without `_id` in the index and you will see the performance difference:

```python
from pprint import pprint
from pymongo import MongoClient, DESCENDING

def print_query_plan(mongo_cursor):
    explanation = mongo_cursor.explain()
    winning_plan = explanation['queryPlanner']['winningPlan'];
    if winning_plan:
        # Query starts with an index scan
        pprint(winning_plan)
    else:
        # Slow query - collection scan
        pprint(explanation['queryPlanner'])

conn = MongoClient()
old_runs_col = conn['fishtest_new']['old_runs']
print_query_plan(old_runs_col.find({
  'finished': True,
  'args.tc': {'$regex': '^([4-9][0-9])|([1-9][0-9][0-9])'}
}, sort=[('_id', DESCENDING)]))
```

With `_id` in the index, the query is fast (entries fetched already-sorted from index):
`IXSCAN -> SORT_KEY_GENERATOR -> SORT -> FETCH`

Without `_id` in the index, the input stages in the query plan is slow (fetched, then sorted):
`IXSCAN -> FETCH -> SORT_KEY_GENERATOR -> SORT`

Reply:
@ppigazzini you can try benchmarking with and without `_id` on this index, and you will find that with `_id` on the index, you will see a significant performance boost 😉

Reply:
Thanks for looking into this. I am a bit allergic to indices which are used infrequently, because they need to be updated and use space (disk/memory), but these are probably fine.